impl/maven-core/src/main/java/org/apache/maven/internal/impl/DefaultProject.java

```diff
@@ -165,6 +166,46 @@ public Optional<Project> getParent() {
         return Optional.ofNullable(session.getProject(parent));
     }
 
+    @Override
+    @Nonnull
+    public List<Profile> getProfiles() {
+        return getModel().getProfiles();
+    }
+
+    @Override
+    @Nonnull
+    public List<Profile> getAllProfiles() {
+        List<org.apache.maven.model.Profile> activeProfiles = new ArrayList<>();
+        for (MavenProject project = this.project; project != null; project = project.getParent()) {
+            activeProfiles.addAll(project.getModel().getProfiles());
+        }
+        return activeProfiles.stream()
+                .map(org.apache.maven.model.Profile::getDelegate)
+                .toList();
+    }
+
+    @Override
+    @Nonnull
+    public List<Profile> getActiveProfiles() {
+        List<org.apache.maven.model.Profile> activeProfiles =
+                project.getActiveProfiles() != null ? project.getActiveProfiles() : List.of();
+        return activeProfiles.stream()
+                .map(org.apache.maven.model.Profile::getDelegate)
+                .toList();
+    }
+
+    @Override
+    @Nonnull
+    public List<Profile> getAllActiveProfiles() {
+        List<org.apache.maven.model.Profile> activeProfiles = new ArrayList<>();
+        for (MavenProject project = this.project; project != null; project = project.getParent()) {
+            activeProfiles.addAll(project.getActiveProfiles());
+        }
+        return activeProfiles.stream()
+                .map(org.apache.maven.model.Profile::getDelegate)
+                .toList();
+    }
+
```

Comment:
You could make the methods a one liner, if you'd like with something like this I think:

```java
    return Stream.iterate(this.project, Objects::nonNull, MavenProject::getParent)
                 .flatMap(project -> project.getActiveProfiles().stream())
                 .map(org.apache.maven.model.Profile::getDelegate)
                 .toList();
```